Server/VikingVault.Services/AccountService.cs

```diff
@@ -2,26 +2,74 @@
 using System.Linq;
 using System.Collections.Generic;
 using System.Text;
+using System.IdentityModel.Tokens.Jwt;
 using VikingVault.DataAccess;
 using VikingVault.Services.Abstractions;
 using VikingVault.DataAccess.Models;
+using Microsoft.Extensions.Primitives;
 
 namespace VikingVault.Services
 {
     public class AccountService:IAccountService
     {
         private readonly VikingVaultDbContext _dbContext;
-        private IList<UserAccount> accountsList;
 
         public AccountService(VikingVaultDbContext dbContext)
         {
             _dbContext = dbContext;
-            accountsList = new List<UserAccount>();
         }
 
-        public UserAccount FindById()
+        public UserAccount GetUserAccount(StringValues token)
         {
-            return null;
+
+            int userId = FindUserIDByToken(token);
+            User returnedUser = _dbContext.User.SingleOrDefault(u => u.Id == userId);
+            Card card = _dbContext.Cards.SingleOrDefault(u => u.UserId == userId);
```

Comment:
What happens if returnedUser is null? you should retrieve card entity from db only if user is not null, otherwise u.UserId will throw an exception

Reply:
Returned user cannot ever be null, as this function is not called, if an user is not logged in

Reply:
What about if you test it from Postman, send a token for which the userId is not in db?

Reply:
It would just throw some exception from the token security, I shall catch it